pkgs/development/python-modules/playwright/update.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl gnused gawk nix-prefetch common-updater-scripts
+
+set -euo pipefail
+
+# This is the target version of playwright-python we want to update to. Find out
+# what's the current one here:
+# https://pypi.org/project/playwright/
+TARGET_VERSION="1.25.1"
+
+# The driver version required for the specific Playwright version we have is
+# hard-coded here:
+# https://github.com/microsoft/playwright-python/blob/v1.25.1/setup.py#L33
+# Most of the time, this should be the latest stable release of the Node-based
+# Playwright version, but that isn't a guarantee, so this needs to be specified
+# as well:
+DRIVER_VERSION="1.25.0"
```

Comment:
This should really be set automatically.

Reply:
One hacky way to get the version from the source repo:

```sh
#!/usr/bin/env nix-shell
#!nix-shell -i bash -p curl gnused gawk python3 python3Packages.wheel python3Packages.setuptools nix-prefetch common-updater-scripts

TARGET_VERSION="1.26.1"

curl -LO "https://github.com/microsoft/playwright-python/archive/refs/tags/v${TARGET_VERSION}.tar.gz"
tar xfz "v${TARGET_VERSION}.tar.gz"
cd "playwright-python-${TARGET_VERSION}"
# Mock out the setup function from setuptools
echo 'def setup(**attrs): pass' > setuptools.py
DRIVER_VERSION=$(python -c 'import setup; print(setup.driver_version)')
cd -

echo $DRIVER_VERSION
```

Gives "1.26.1" at the end when I run this.

Reply:
Can we parse that we sed/grep from https://github.com/microsoft/playwright-python/blob/8f55a0ee3df97e6297f52147213c75eea51b12fd/setup.py#L33 instead of execing python?

Reply:
Yes, a grep-sed version:

```sh
#!/usr/bin/env nix-shell
#!nix-shell -i bash -p curl gnugrep gnused gawk nix-prefetch common-updater-scripts

TARGET_VERSION="1.26.1"

setup_py_url="https://github.com/microsoft/playwright-python/raw/v${TARGET_VERSION}/setup.py"

DRIVER_VERSION=$(curl -L "$setup_py_url" | grep '^driver_version =' | sed $'s/^driver_version\s*=\s*[\'"]\([^\'\"]*\)[\'"].*/\\1/')

echo $DRIVER_VERSION
```

Reply:
That looks nice.